.gitattributes

```diff
@@ -0,0 +1,2 @@
+components/**/*.hex binary
```

Comment:
bit confused, a .hex file is clearly not a binary file

Reply:
Do you care about the line endings of the hexfile?

Do you have a better way to exclude `components` from the compliance checks? The alternative is to remove the entire `GitDiffCheck` check, which seems more intrusive. We could do that since whitespace is already checked by checkpatch, so the check likely has limited value.